File: CountingBits3.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> countBits(int num) {
        
        vector<int> bits(num + 1, 0);
        
        for(int i = 1; i <= num; i++) bits[i] = bits[i & (i - 1)] + 1;
		
		return bits;
        
    }
};

int main() {
	
	int num;
	cin>>num;
	
	Solution *solution = new Solution();
	vector<int> result = solution->countBits(num);
	
	for(int i = 0; i < result.size(); i++) {
		cout<<result[i]<<"  ";
	}
	
	return 0;
	
}

File: CountingBits3.py
class Solution:
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        bits = [0] * (num + 1)
        for i in range(1, num + 1):
            bits[i] = bits[i & (i - 1)] + 1
        return bits


if __name__ == '__main__':
    num = int(input())
    solution = Solution()
    print(solution.countBits(num))
